projects/igniteui-angular/src/lib/navigation-drawer/themes/_base.scss

```diff
@@ -0,0 +1,265 @@
+            inset-inline-start: 0;
+            z-index: 999;
+            transition: width, padding, transform;
+            transition-timing-function: $ease-in-out-quad;
+            box-shadow: var-get($_theme, 'elevation');
+            padding: var(--aside-padding);
+            border-inline-end: rem(1px) solid var-get($_theme, 'border-color');
+            border-radius: var-get($_theme, 'border-radius');
+
+            &.panning {
+                overflow-x: hidden;
+                transition: none;
+            }
+        }
+
+        @include e(aside, collapsed) {
+            transform: translate3d(-300px, 0, 0);
+
+            [dir='rtl'] & {
+                transform: translate3d(300px, 0, 0);
+            }
+
+            box-shadow: none;
+
+            &.igx-nav-drawer__aside--right {
+                transform: translate3d(300px, 0, 0);
+                box-shadow: none;
+
+                [dir='rtl'] & {
+                    transform: translate3d(-300px, 0, 0);
+                }
+            }
+        }
+
+        @include e(aside, right) {
+            inset-inline-start: auto;
+            inset-inline-end: 0;
+            border-inline-end: none;
+            border-inline-start: rem(1px) solid var-get($_theme, 'border-color');
+        }
+
+        @include e(aside, mini) {
+            transition-duration: .3s;
+            width: #{var(--igx-nav-drawer-size--mini)};
+            min-width: fit-content;
+
+            %item {
+                justify-content: center;
+                min-width: fit-content;
+
+                igx-icon {
+                    margin-inline-start: 0;
+                }
+            }
+        }
+
+        @include e(aside, normal) {
+            transition-duration: .3s;
+            width: var(--igx-nav-drawer-size);
+        }
+
+        @include e(aside, pinned) {
+            position: relative;
+            box-shadow: none;
+            z-index: 0;
+
+            &.igx-nav-drawer__aside--collapsed {
+                transform: none;
+                width: 0;
+                overflow: hidden;
+                border: none;
+                padding: 0;
+            }
+        }
+
+        @include e(overlay) {
+            opacity: 1;
+            background: color(null, 'gray', 500, .54);
+            transition: opacity, visibility;
+            transition-duration: .3s, .3s;
+            transition-timing-function: ease-in, step-start;
+            transition-delay: 0s, 0s;
+            position: absolute;
+            inset-inline-start: 0;
+            top: 0;
+            width: 100%;
+            height: 100%;
+            visibility: visible;
+            z-index: 999;
+
+            &.panning {
+                transform: translate3d(0, 0, 0);
+                transition: none;
+
+                &.igx-nav-drawer__overlay--hidden {
+                    /* must be visible during pan.. */
+                    visibility: visible;
+                }
+            }
```

Comment:
This is messy. I guess the `.panning` class should rather be renamed to `.igx-nav-drawer__overlay--panning`. Using the BEM sass mixins then you can add the style rules using the following method:

```scss
@include e(overlay, $m: panning) {
  transform: translate3d(0, 0, 0);
  transition: none;
}

@include e(overlay, $mods: (panning, hidden) {
  visibility: visible;
}
```